src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp

```diff
@@ -745,6 +764,34 @@ NetworkDescription LevelZeroCompilerInDriver<TableExtension>::compileIR(const st
                     ". ",
                     getLatestBuildError());
 
+    auto getMemoryUsage = []() -> size_t {
+        std::ifstream procStatus("/proc/self/status");
+        std::string line;
+        while (std::getline(procStatus, line)) {
+            if (line.substr(0, 6) == "VmRSS:") {
+                std::istringstream iss(line);
+                std::string key;
+                size_t memoryUsageKb;
+                iss >> key >> memoryUsageKb;
+                return memoryUsageKb * 1024;  // Convert from KB to bytes
+            }
+        }
+        return 0;  // Return 0 if VmRSS is not found (should not happen on Linux)
+    };
+
+    printf("\n\nDEBUG Before release memory ! \n");
+    size_t beforeClearMemoryUsage = getMemoryUsage();
+    std::cout << "Memory usage before clearing serializedIR: " << beforeClearMemoryUsage << " bytes" << std::endl;
+    print_memory_usage();
+
+    serializedIR.clear();
+    std::vector<uint8_t>().swap(serializedIR); // deallocates the memory
```

Comment:
This looks tricky, but shrink_to_fit is only introduced in C++20.
It might be cleaner to use scope to free the temp object. Let's consider putting serializeIR/createGraphin separate function.

Reply:
thanks ! 
ya, use scope to free the temp object might be the better approach, i will work on it